prow/plugins/verify-owners/verify-owners.go

```diff
@@ -0,0 +1,151 @@
+}
+
+func handlePullRequest(pc plugins.PluginClient, pre github.PullRequestEvent) error {
+	if pre.Action != github.PullRequestActionOpened && pre.Action != github.PullRequestActionReopened && pre.Action != github.PullRequestActionSynchronize {
+		return nil
+	}
+	return handle(pc.GitHubClient, pc.GitClient, pc.Logger, &pre, pc.PluginConfig.Owners.LabelsBlackList)
+}
+
+func handle(ghc githubClient, gc *git.Client, log *logrus.Entry, pre *github.PullRequestEvent, labelsBlackList []string) error {
+	org := pre.Repo.Owner.Login
+	repo := pre.Repo.Name
+	wrongOwnersFiles := map[string]error{}
+
+	// Get changes.
+	changes, err := ghc.GetPullRequestChanges(org, repo, pre.Number)
+	if err != nil {
+		return fmt.Errorf("error getting PR changes: %v", err)
+	}
+
+	// List modified OWNERS files.
+	var modifiedOwnersFiles []string
+	for _, change := range changes {
+		if filepath.Base(change.Filename) == ownersFileName {
+			modifiedOwnersFiles = append(modifiedOwnersFiles, change.Filename)
+		}
+	}
+	if len(modifiedOwnersFiles) == 0 {
+		return nil
+	}
+
+	// Clone the repo, checkout the PR.
+	r, err := gc.Clone(pre.Repo.FullName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.CheckoutPullRequest(pre.Number); err != nil {
+		return err
+	}
+
+	// Check each OWNERS file.
+	for _, f := range modifiedOwnersFiles {
+		// Try to load OWNERS file.
+		simple, err := repoowners.ParseSimpleConfig(filepath.Join(r.Dir, f))
+		if err != nil {
+			wrongOwnersFiles[f] = fmt.Errorf("error occurred parsing the OWNERS file in %s: %v", f, err)
+			break
+		}
+		// Check labels against blacklist
+		if sets.NewString(simple.Config.Labels...).HasAny(labelsBlackList...) {
+			wrongOwnersFiles[f] = fmt.Errorf("OWNERS file contains blacklisted labels: %v", sets.NewString(simple.Config.Labels...).Intersection(sets.NewString(labelsBlackList...)))
+			break
+		}
+		// Check approvers isn't empty
+		if filepath.Dir(f) == "." && len(simple.Config.Approvers) == 0 {
+			wrongOwnersFiles[f] = errors.New("no approvers defined for the repo")
+			break
+		}
+	}
+	// React if we saw something.
+	if len(wrongOwnersFiles) > 0 {
+		log.Debugf("Adding %s label.", invalidOwnersLabel)
+		if err := ghc.AddLabel(org, repo, pre.Number, invalidOwnersLabel); err != nil {
+			return err
+		}
+		log.Debugf("Creating a review for %d OWNERS files.", len(wrongOwnersFiles))
+		var comments []github.DraftReviewComment
+		for errFile, errMsg := range wrongOwnersFiles {
+			resp := fmt.Sprintf("Invalid OWNERS file: %v", errMsg)
+			comments = append(comments, github.DraftReviewComment{
+				Path: errFile,
+				Body: resp,
+			})
+		}
+		err := ghc.CreateReview(org, repo, pre.Number, github.DraftReview{
+			Action:   github.Comment,
+			Comments: comments,
+		})
+		if err != nil {
+			log.WithError(err).Error("Error creating a review for OWNERS file.")
```

Comment:
This error should be returned instead of just logging since it is an error in the handling of the event:
```go
return fmt.Errorf("error creating a review for invalid OWNERS file(s): %v", err)
```